keycloak-init-container/extensions.sh

```diff
@@ -1,52 +1,63 @@
-#!/bin/bash -e
+#!/bin/bash
 
-EXTENSIONS_VOLUME="${EXTENSIONS_VOLUME:-/opt/extensions}"
+shopt -so errexit pipefail
 
-mkdir -p "$EXTENSIONS_VOLUME"
-cd "$EXTENSIONS_VOLUME"
-echo "Target directory: $EXTENSIONS_VOLUME"
+## External variables used by this script
+declare -rx KEYCLOAK_EXTENSIONS
+declare -rx EXTENSIONS_VOLUME="${EXTENSIONS_VOLUME:-/opt/extensions}"
 
 # Download a single extension
 download_extension() {
-    local EXTENSION_URL="$1"
+    local extension_url="$1"
 
-    if [[ -z "$EXTENSION_URL" ]]; then
+    if [[ -z "$extension_url" ]]; then
         return
     fi
-    echo 
-    echo "Downloading extension from $EXTENSION_URL"
-    local CURL_COMMAND="$(curl --verbose --location  --remote-name --remote-header-name --write-out "%{http_code} %{filename_effective}"  --silent "$EXTENSION_URL" 2> /tmp/headers)"
-  
-    local STATUS_CODE=${CURL_COMMAND:0:3}
-
-    if [ $STATUS_CODE -eq "200" ]; then
-        local FILENAME=${CURL_COMMAND:4}
+    echo
+    echo "Downloading extension from $extension_url"
+    local curl_command
+    curl_command="$(curl --verbose --location --remote-name --remote-header-name --write-out "%{http_code} %{filename_effective}" --silent "$extension_url" 2> /tmp/headers)"
+
+    local status_code=${curl_command:0:3}
+
+    if [[ "$status_code" -eq "200" ]]; then
+        local filename=${curl_command:4}
         echo "Extension downloaded successfully"
 
         # Try to get the filename from the response headers and return
         # a random name if that fails
-        if ! grep -q -i '^< content-disposition:.*filename=' /tmp/headers ; then
-            local F="$(od -N8 -tx1 -An -v /dev/urandom | tr -d "").jar"
-            mv "$FILENAME" "$F"
-            FILENAME="$F"
+        if ! grep -q -i '^< content-disposition:.*filename=' /tmp/headers; then
+            local F
+            F="$(od -N8 -tx1 -An -v /dev/urandom | tr -d "").jar"
+            mv "$filename" "$F"
+            filename="$F"
         fi
-        echo " --> $FILENAME"
+        echo " --> $filename"
     else
-        echo -e "Can not download the extension: $EXTENSION_URL\nError code: $STATUS_CODE"
-        ((STATUS+=1))
+        echo -e "Can not download the extension: $extension_url\nError code: $status_code"
+        ((FAILED_DOWNLOADS+=1))
+        return 1
     fi
-
 }
 
+mkdir -p "$EXTENSIONS_VOLUME"
+if ! cd "$EXTENSIONS_VOLUME"; then
+    echo "Cannot cd to $EXTENSIONS_VOLUME"
+    exit 1
+fi
+echo "Target directory: $EXTENSIONS_VOLUME"
+
 # Parse the environment variable and download the extensions from the list
 IFS=,
-STATUS=0
-for EXT in ${KEYCLOAK_EXTENSION[@]} ; do
-    download_extension "$EXT"
+FAILED_DOWNLOADS=0
+
+for EXT in ${KEYCLOAK_EXTENSIONS[*]}; do
```

Comment:
This is where `set -u` above would help to catch this kind of variable names typo.